### PR TITLE
Fix/apps review

### PR DESCRIPTION
**Pull request by lmichel**
Answer to the APPs review
- clarification on the condition for ATTRIBUTE to have a null value
- some though about the API design
- clarification on the issue with on the fly annotation in TAP service

### CODE REVIEW

File: doc/api.tex [hunk 1]
```diff
@@ -5,8 +5,20 @@
...
+  \item Model classes: For the most popular models and especially for the models that provide reusable 
+        components (Meas/Coord, PhotDM), the API can include objects instantiating model classes (e.g. Photometric filter).
+  \item Automatic extraction of class instances of common frameworks: One of the goal of the data annotation is to 
+        facilitate the connection between data and API code. This could be achieved with an API able to automatically 
+        build objects from the annotated data without asking the user to infer on meta data. Taking the Astropy example, 
+        a model-enable Python API should be able to build e.g. \texttt{Skycoord} instances in a transparent way and 
+        feed them with \texttt{FIELD} values from \texttt{TABLE} rows or from \texttt{PARAM}.
  \end{enumerate}
 
+These different levels should be hidden to the end user. Using MIVOT annotations makes sense only if the client code provides
```
Comment (andamian) [inline doc/api.tex:19]: I would say that for practical reasons level 1 and 2 are not very useful and client software should provide level 3 and 4 APIs to end users.

Comment (lmichel) [reply doc/api.tex]: I reworded the paragraph,

File: doc/api.tex [hunk 2]
```diff
@@ -16,8 +16,11 @@
         feed them with \texttt{FIELD} values from \texttt{TABLE} rows or from \texttt{PARAM}.
  \end{enumerate}
 
-These different levels should be hidden to the end user. Using MIVOT annotations makes sense only if the client code provides 
+Level 1 and 2 may only be useful for client developpers, they should be hidden to the end user. 
+Using MIVOT annotations makes sense only if the client code provides 
 the users with a direct access to both modeled quantities and links that connect them. 
+This can be achieved by providing level 3 and 4 APIs to end users.
```
Comment (andamian) [inline doc/api.tex:22]: Yes, this looks better. I would say that MIVOT is a machine level system that relies primarily on higher level APIs to provide representations of the models (level 3 and 4) to end users

**Review by andamian — DISMISSED** [review]

File: doc/api.tex (hunk 2 above)
Comment (lmichel) [reply doc/api.tex]: I've changed the text with your wording

**Pull request merged**